nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java

```diff
@@ -0,0 +1,273 @@
+            return cachedMatchingPredicate;
+        }
+
+        final MatchingStrategy matchingStrategy = getMatchingStrategy(context);
+
+        switch (matchingStrategy) {
+            case ENUMERATION:
+                return determineMatchingPredicateBasedOnEnumeration(context, flowFile);
+            case REGEX:
+                return determineMatchingPredicateBasedOnRegex(context, flowFile);
+            default:
+                throw new IllegalArgumentException(
+                        "Cannot determine matching predicate for unsupported strategy " + matchingStrategy + " !"
+                );
+        }
+    }
+
+    private static Predicate<String> determineMatchingPredicateBasedOnEnumeration(ProcessContext context, FlowFile flowFile) {
+        final String attributeSetDeclaration = getAttributeSet(context, flowFile);
+        final String delimiter = getDelimiter();
+
+        Set<String> attributeSet = Arrays.stream(attributeSetDeclaration.split(Pattern.quote(delimiter)))
+                .map(String::trim)
+                .filter(attributeName -> !attributeName.trim().isEmpty())
+                .collect(Collectors.toSet());
+
+        return attributeSet::contains;
+    }
+
+    private static Predicate<String> determineMatchingPredicateBasedOnRegex(ProcessContext context, FlowFile flowFile) {
+        Pattern attributeRegex = getAttributeRegex(context, flowFile);
+
+        return attributeName -> attributeRegex.matcher(attributeName).matches();
+    }
+
+    /* properties */
+
+    private static FilterMode getFilterMode(ProcessContext context) {
+        final String rawFilterMode = context
+                .getProperty(FILTER_MODE)
+                .getValue();
+
+        if (FILTER_MODE_VALUE_REMOVE.getValue().equals(rawFilterMode)) {
+            return FilterMode.REMOVE;
+        }
+        return FilterMode.RETAIN;
+    }
+
+    private static MatchingStrategy getMatchingStrategy(ProcessContext context) {
+        final String rawMatchingStrategy = context
+                .getProperty(MATCHING_STRATEGY)
+                .getValue();
+
+        if (MATCHING_STRATEGY_VALUE_REGEX.getValue().equals(rawMatchingStrategy)) {
+            return MatchingStrategy.REGEX;
+        }
+        return MatchingStrategy.ENUMERATION;
+    }
+
+    private static String getAttributeSet(ProcessContext context, FlowFile flowFile) {
+        return context.getProperty(ATTRIBUTE_SET).evaluateAttributeExpressions(flowFile).getValue();
+    }
+
+    private static String getDelimiter() {
+        return DELIMITER_VALUE;
+    }
+
+    private static Pattern getAttributeRegex(ProcessContext context, FlowFile flowFile) {
+        return Pattern.compile(
+                context.getProperty(ATTRIBUTE_REGEX).evaluateAttributeExpressions(flowFile).getValue()
+        );
+    }
+
+    private enum FilterMode {
+        RETAIN,
+        REMOVE,
+    }
+
+    private enum MatchingStrategy {
+        ENUMERATION,
+        REGEX,
```

Comment:
Recommend using `PATTERN` here instead of `REGEX` to align with Java Pattern object, and adjusted property name.